monkey/tests/unit_tests/common/test_agent_configuration.py

```diff
@@ -0,0 +1,230 @@
+    assert config.inaccessible_subnets == INACCESSIBLE_SUBNETS
+    assert config.local_network_scan == LOCAL_NETWORK_SCAN
+    assert config.subnets == SUBNETS
+
+
+TIMEOUT = 2.525
+ICMP_CONFIGURATION = {"timeout": TIMEOUT}
+
+
+def test_icmp_scan_configuration_schema():
+    schema = ICMPScanConfigurationSchema()
+
+    config = schema.load(ICMP_CONFIGURATION)
+
+    assert config.timeout == TIMEOUT
+
+
+PORTS = [8080, 443]
+
+TCP_SCAN_CONFIGURATION = {"timeout": TIMEOUT, "ports": PORTS}
+
+
+def test_tcp_scan_configuration_schema():
+    schema = TCPScanConfigurationSchema()
+
+    config = schema.load(TCP_SCAN_CONFIGURATION)
+
+    assert config.timeout == TIMEOUT
+    assert config.ports == PORTS
+
+
+FINGERPRINTERS = [{"name": "mssql", "options": {}}]
+NETWORK_SCAN_CONFIGURATION = {
+    "tcp": TCP_SCAN_CONFIGURATION,
+    "icmp": ICMP_CONFIGURATION,
+    "fingerprinters": FINGERPRINTERS,
+    "targets": SCAN_TARGET_CONFIGURATION,
+}
+
+
+def test_network_scan_configuration():
+    schema = NetworkScanConfigurationSchema()
+
+    config = schema.load(NETWORK_SCAN_CONFIGURATION)
+
+    assert config.tcp.ports == TCP_SCAN_CONFIGURATION["ports"]
+    assert config.tcp.timeout == TCP_SCAN_CONFIGURATION["timeout"]
+    assert config.icmp.timeout == ICMP_CONFIGURATION["timeout"]
+    assert config.fingerprinters[0].name == FINGERPRINTERS[0]["name"]
+    assert config.fingerprinters[0].options == FINGERPRINTERS[0]["options"]
+    assert config.targets.blocked_ips == BLOCKED_IPS
+    assert config.targets.inaccessible_subnets == INACCESSIBLE_SUBNETS
+    assert config.targets.local_network_scan == LOCAL_NETWORK_SCAN
+    assert config.targets.subnets == SUBNETS
+
+
+def test_exploitation_options_configuration_schema():
+    ports = [1, 2, 3]
+    schema = ExploitationOptionsConfigurationSchema()
+
+    config = schema.load({"http_ports": ports})
+
+    assert config.http_ports == ports
+
+
+def test_exploiter_configuration_schema():
+    name = "bond"
+    options = {"gun": "Walther PPK", "car": "Aston Martin DB5"}
+    supported_os = [OperatingSystems.LINUX, OperatingSystems.WINDOWS]
+    schema = ExploiterConfigurationSchema()
+
+    config = schema.load(
+        {"name": name, "options": options, "supported_os": [os_.name for os_ in supported_os]}
+    )
+
+    assert config.name == name
+    assert config.options == options
+    assert config.supported_os == supported_os
+
+
+BRUTE_FORCE = [
+    {"name": "ex1", "options": {}, "supported_os": ["LINUX"]},
+    {
+        "name": "ex2",
+        "options": {"smb_download_timeout": 10},
+        "supported_os": ["LINUX", "WINDOWS"],
+    },
+]
+VULNERABILITY = [
+    {
+        "name": "ex3",
+        "options": {"smb_download_timeout": 10},
+        "supported_os": ["WINDOWS"],
+    },
+]
+EXPLOITATION_CONFIGURATION = {
+    "options": {"http_ports": PORTS},
+    "brute_force": BRUTE_FORCE,
+    "vulnerability": VULNERABILITY,
+}
+
+
+def test_exploitation_configuration():
```

Comment:
I'm not sure what we're testing with most of these tests. That mock dictionaries match the Schema? What's the value of these tests?

Reply:
At the present time, they don't provide much value. When we add validation, they'll be useful templates.